components/classes/StudentsPrintOptions.tsx

```diff
@@ -17,21 +19,20 @@ import { toggle } from "radash";
 import { FC } from "react";
 
 /**
- * Options for the Student List Printout.
+ * Options for the Student List Printout. People can choose which columns to
+ * display, how many empty columns to display, and configure other options here.
  *
  * @param form The form control values.
  * @param setForm The form setter.
  * @param formProps The form control props.
- * @param userRole The role of the user visitng the page. Exposes Student ID if the user is a Teacher.
- *
- * @returns A Print Options.
+ * @param user The user visiting the page. Exposes Student ID if the user isn't a Student.
  */
 const StudentsPrintOptions: FC<{
   form: OptionsType;
   setForm: (form: OptionsType) => void;
   formProps: FormControlProps<keyof OptionsType>;
-  userRole: UserRole;
-}> = ({ form, setForm, formProps, userRole }) => {
+  user: User | null;

```

Comment:
Why do we have to concern the possibility that the user is null when the user must log in before using this page?

Reply:
Since this page is ISR, we cannot access the user’s cookies when generating the page server-side. To keep permissions working, we fetch user info client-side instead, but this means that the page generated server-side would have `user` as `null` until the client completes the fetch.

Try logging in as a Teacher—you’ll see the Student ID option pop in a split second from page load. That’s the client-side fetch at work.